python/pyspark/ml/tests.py

```diff
@@ -371,6 +378,103 @@ def test_fit_maximize_metric(self):
         self.assertEqual(1.0, bestModelMetric, "Best model has R-squared of 1")
 
 
+class RegressorTest(PySparkTestCase):
+
+    def setupData(self):
+        try:
+            self.df
+        except AttributeError:
+            from pyspark.mllib.linalg import Vectors
+            sqlContext = SQLContext(self.sc)
+            self.df = sqlContext.createDataFrame([
+                (1.0, Vectors.dense(1.0)),
+                (0.0, Vectors.sparse(1, [], []))], ["label", "features"])
+
+    def test_linear_regression(self):
+        self.setupData()
+        lr = LinearRegression(maxIter=5, regParam=0.0, solver="normal")
+        model = lr.fit(self.df)
+        self.assertEquals(1, model.numFeatures)
+
+    def test_decision_tree_regressor(self):
+        self.setupData()
+        dt = DecisionTreeRegressor(maxDepth=2)
+        model = dt.fit(self.df)
+        self.assertEquals(1, model.numFeatures)
+
+    def test_random_forest_regressor(self):
+        self.setupData()
+        rf = RandomForestRegressor(numTrees=2, maxDepth=2, seed=42)
+        model = rf.fit(self.df)
+        self.assertEquals(1, model.numFeatures)
+
+    def test_gbt_regressor(self):
+        self.setupData()
+        gbt = GBTRegressor(maxIter=5, maxDepth=2)
+        model = gbt.fit(self.df)
+        self.assertEquals(1, model.numFeatures)
+
+
+class ClassificationTest(PySparkTestCase):
+
+    def setupData(self):
+        try:
+            self.df
+        except AttributeError:
+            from pyspark.mllib.linalg import Vectors
```

Comment:
`Vectors` and `StringIndexer` is not used in any other place. It is better not to expand the scope in my though.